### PR TITLE
Add UI changes for form name and slug settings

### PR DESCRIPTION
**Pull request by chrispymm**
* Adds the Form name and url settings screen template 
* Updates the settings index page
  * Adds form name and url 
  * Hides form details 
* Update the create new form modal

All changes are behind the NAME_SLUG feature flag, so this should be good to deploy.

Co-authored by @njseeto 

## With feature flag ENABLED
image

image

image

## Without feature flag
image

image

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -308,6 +308,20 @@ en:
       negative: 'No'
   settings:
     name: 'Settings'
+    form_name:
+      heading: 'Form name and URL'
+      lede: ''
+      help: Additional guidance for URL is available on
+      form_name:
+        label: 'Form name'
+        lede: This appears in the header of all your form pages and is what your form is liseted as in the MoJ Forms editor. A good name describes the task the form is designer for and starts with an action, such as "Get help with..." or "Apply for..." %{href}.
```
Comment (H3ll3m4) [inline config/locales/en.yml:317]: typo listed

Comment (H3ll3m4) [inline config/locales/en.yml:317]: typo designed

**Review by H3ll3m4 — APPROVED** [review]
🚀

**Review by H3ll3m4 — DISMISSED** [review]
🚀

**Review by StevenLeighton21 — APPROVED** [review]

**Pull request merged**